examples/02_decoding/plot_haxby_full_analysis.py

```diff
@@ -44,49 +45,53 @@
 # extract tags indicating to which acquisition run a tag belongs
 session_labels = labels["chunks"][task_mask]
 
+# apply the task_mask to  fMRI data (func_filename)
+from nilearn.image import index_img
+task_data = index_img(func_filename, task_mask)
 
 ##########################################################################
-# Then we use scikit-learn for decoding on the different masks
-# -------------------------------------------------------------
-# The classifier: a support vector classifier
-from sklearn.svm import SVC
-classifier = SVC(C=1., kernel="linear")
-
-# A classifier to set the chance level
-from sklearn.dummy import DummyClassifier
-dummy_classifier = DummyClassifier()
+# Decoding on the different masks
+# --------------------------------
+#
+# The classifier used here is a support vector classifier (svc). We use
+# class:`nilearn.decoding.Decoder` and specify the classifier.
+import numpy as np
+from nilearn.decoding import Decoder
 
 # Make a data splitting object for cross validation
 from sklearn.model_selection import LeaveOneGroupOut, cross_val_score
 cv = LeaveOneGroupOut()
 
+# We use :class:`sklearn.dummy.DummyClassifier` as a baseline. 
+from sklearn.dummy import DummyClassifier
+dummy_classifier = DummyClassifier()
+
 mask_names = ['mask_vt', 'mask_face', 'mask_house']
 
 mask_scores = {}
 mask_chance_scores = {}
 
 for mask_name in mask_names:
-    print("Working on mask %s" % mask_name)
+    print("Working on %s" % mask_name)
     # For decoding, standardizing is often very important
     mask_filename = haxby_dataset[mask_name][0]
     masker = NiftiMasker(mask_img=mask_filename, standardize=True)
-    masked_timecourses = masker.fit_transform(
-        func_filename)[task_mask]
-
+    masked_timecourses = masker.fit_transform(func_filename)[task_mask]
     mask_scores[mask_name] = {}
     mask_chance_scores[mask_name] = {}
 
     for category in categories:
         print("Processing %s %s" % (mask_name, category))
         classification_target = (stimuli[task_mask] == category)
-        mask_scores[mask_name][category] = cross_val_score(
-            classifier,
-            masked_timecourses,
-            classification_target,
-            cv=cv,
-            groups=session_labels,
-            scoring="roc_auc",
-        )
+        # Specify the classifier to the decoder object. With the decoder we can
+        # input the masker directly
+        decoder = Decoder(estimator='svc_l1', cv=cv,
```

Comment:
I believe that we are using the l1 here because it is intra subject. Maybe it would be useful to mention it here.